classes/class09/README.md

```diff
@@ -1 +1,179 @@
+draw.io is a cool web-based application that offers a lot of nice tools for making process diagrams and flow charts like the one below, check it out!
+
+![Example diagram from draw.io](images/draw_example.png)
+
+We aren't going to require any one specific type of diagram for your projects in this class.
+Instead, we're going to recommend a couple of different options that you might consider diagramming for managing complexity in your application.
+Let's look at things you might consider diagramming.
+
+####Logic Flow
+if your application involves a lot of different routes and middleware, you might consider mapping how a request moves through your application.
+As an example, let's look at how we might diagram a simple application with a few routes and a few pieces of middleware.
+![Example application logic flow](images/app_flow.png)
+If we follow a request in from the user, we can clearly see how it moves through the router to a route, which routes need to access our database, and which ones redirect.
+You can pick and choose which components inside your application you choose to represent, but make sure that the amount of information you show is useful.
+Too much information and the diagram might be difficult to follow, too little and it might not be useful, or could even be misleading.
+
+####Interactions
+If you anticipate having a lot of different interfaces you might want to create a diagram that represents what the user will see happen in response to certain actions.
+You might find it easier to draw these out by hand at first, or use a tool like [balsamiq](https://balsamiq.com/) or [inVision](http://www.invisionapp.com/), but don't think you need a high resolution representation to make something of value.
+![](http://www.seanlemme.com/assets/img/portfolio/nurse2.png)
+*Image via seanlemme.com*
+
+####Network Layout
+If you ever get to the point that you're working on a *really* big application that's distributed across multiple different severs, you'll need to start diagramming a map of all of your different servers, databases, and other networked components, and how they all connect.
+We'll never get around to anything remotely close to the system shown below (in fact I'd be surprised if anyone has a configuration like that), but the principle is still there.
+![AWS network layout](images/AWSExample.png)
+
+##Refactoring
+As you develop (or let's be honest - have been developing), you might find yourself doing hacky things just to get your code to work.
+Avoid this if you can.
+However if you do, it's valuable to take a few minutes to **refactor** your code.
+If you're unfamiliar with this term, refactoring basically refers to restructuring how your code is laid out across files, functions, modules, etc. without necessarily modifying any functionality.
+In order for you to be effective at refactoring, let's go through a typical folder structure for one of our applications and look at what goes where.
```

Comment:
Its probably worth adding a bit more about refactoring code in terms of how functions/objects/classes interact. The current description makes it sound like its mostly just what files and modules code chunks are placed in.